platform/api/esx/api.go

```diff
@@ -429,6 +429,12 @@ func (a *API) CreateDevice(name string, conf *conf.Conf, ips *IpPair) (*ESXMachi
 	}
 
 	if ips != nil {
+		// based on: https://mirrors.edge.kernel.org/pub/linux/utils/boot/dracut/dracut.html#_network
+		kArgs := fmt.Sprintf("ip=%s::%s:%d:localhost:ens192:off:1.1.1.1:1.0.0.1", ips.Public, ips.PublicGw, ips.SubnetSize)
```

Comment:
I think instead of SubnetSize the netmask is expected: https://www.flatcar.org/docs/latest/provisioning/ignition/network-configuration/#using-static-ip-addresses-with-ignition
I would also leave the hostname empty instead of using `localhost`.
Is there a difference between `off` and `none` (our docs only say `none`)?

Reply:
I gave a try with both. Also seen in this post: https://www.redhat.com/sysadmin/network-confi-initrd  - but can't find an actual documentation mentioning the expected format. I think kernel can deal with both CIDR notation and netmask ? Initially I was calculating the netmask: https://github.com/flatcar-linux/mantle/compare/1d654413912be9aba3c24a04303b445b76cfdc4e..e175cbb3d76f19f5557f119140dc8e4e9964f619

About `none/off` they seem to have the same effect (static IP) (https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/networking_guide/sec-configuring_ip_networking_from_the_kernel_command_line)

Reply:
The only thing that counts is the code in this script which may divert from other implementations: https://github.com/flatcar-linux/bootengine/blob/flatcar-master/dracut/03flatcar-network/parse-ip-for-networkd.sh#L41
and indeed the `elif` case seems to accept an integer (SubnetSize) instead of a netmask. And `none` is also the same as `off` in our case: https://github.com/flatcar-linux/bootengine/blob/flatcar-master/dracut/03flatcar-network/parse-ip-for-networkd.sh#L53

Reply:
```suggestion
		kArgs := fmt.Sprintf("ip=%s::%s:%d::ens192:off:1.1.1.1:1.0.0.1", ips.Public, ips.PublicGw, ips.SubnetSize)
```
So I would just leave out setting the hostname explicitly to localhost.